### PR TITLE
Support regex path matching for autoLogging.ignorePaths option

### PR DESCRIPTION
**Pull request by jafern14**
I was using the latest version and have a use-case where I want to ignore a dynamic range of paths that can change over time. In order to support this I thought that adding RegExp support to the ignoring mechanism might be an option.

Let me know what you think. Thanks for taking a look.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "pino-http",
-  "version": "5.1.0",
+  "version": "5.2.0",
```
Comment (mcollina) [inline package.json:3]: Please rever this.

File: logger.js [hunk 2]
```diff
@@ -100,7 +100,9 @@ function pinoLogger (opts, stream) {
           url = URL.parse(req.url)
         }
         if (url && url.pathname) {
-          shouldLogSuccess = !autoLoggingIgnorePaths.includes(url.pathname)
+          shouldLogSuccess = !autoLoggingIgnorePaths.find(ignorePath => {
+            return new RegExp(ignorePath).test(url.pathname)
+          })
```
Comment (mcollina) [inline logger.js:105]: We should not be generating a new regexp for strings that are anot already regexp.

Comment (jafern14) [reply logger.js:109]: @mcollina let me know if you prefer some other approach. I thought this would work as well and be a bit more explicit.

**Comment by mcollina** [comment]
@jsumners @davidmarkclements wdyt?

File: logger.js (hunk 2 above)
Comment (jsumners) [reply logger.js:109]: This is taking user input (`url.pathname`) and creating a new regular expression from it. See https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS

Comment (mcollina) [reply logger.js:109]: This has been already fixed.

Comment (katlimruiz) [reply logger.js:109]: 👍

**Comment by jsumners** [comment]
ping: @davidmarkclements

**Comment by DanHulton** [comment]
Is there any intention of merging this?  I can always target a branch to get this functionality, but it would be a lot cleaner to just be able to use the base package.

I'm not clear on the governance of this project, does @davidmarkclements need to +1 it, because it's been many months waiting on him to check in.

**Review by davidmarkclements — APPROVED** [review]
lgtm

**Comment by davidmarkclements** [comment]
published as 5.3.0 sorry for the delay everyone

**Pull request merged**